drivers/counter/counter_ll_stm32_timer.c

```diff
@@ -470,6 +470,9 @@ static int counter_stm32_init_timer(const struct device *dev)
 	}
 	data->freq = tim_clk / (cfg->prescaler + 1U);
 
+	/* Reset timer to default state using RCC */
+	LL_TIM_DeInit(timer);
```

Comment:
If we're to add device reset, which will eventually end up in most drivers, I think the clean way is to add include/zephyr/drivers/reset.h support.
I acknowledge this is a significantly larger change though...

Reply:
Could you confirm that I understand it correctly:
- I need to add reset driver for STM32 and modify DTS for all STM32 devices appropriately.
- In the driver I will use RCC to reset requested module.
- Use reset api in UART and timer (and modify their DTS, I guess).

I agree that this approach is much better, cleaner in a long term. If I have any questions I'll let you know.

Reply:
Your understanding is correct. This is a significantly larger change (but relatively simple).

Reply:
I added reset drivers and used reset API in UART and timer drivers. It was **significant** amount of work. I wouldn't say it was simple, but definitely worth it.